### PR TITLE
feat: enable disabled actions by feature flags

### PR DESCRIPTION
**Pull request by pnal**
This PR aims to add ability to enable disabled action by feature flag.

How to test:
```php
return new oat\tao\model\action\ActionBlackList(array(
    'disabledActions' => array(
        'action-name'
    ),
    'enabledActionsByFeatureFlagMap' => array(
       'action-name' => 'FEATURE_FLAG_THAT_ENABLES_ACTION_NAME'
    )
));
```
- Add some action id in a list of `disabledActions` in  `config/tao/ActionBlackList.conf.php` as above
- Validate that action button is **not visible** in TAO3.x UI (existing functionality)
- Add a map pair into `enabledActionsByFeatureFlagMap` in the same conf file
- Set the chosen environment variable to `true`; validate that action button is **visible**
- Set the chosen environment variable to `false`; validate that action button is **not visible**

### CODE REVIEW

File: models/classes/action/ActionBlackList.php [hunk 1]
```diff
@@ -33,16 +33,31 @@ class ActionBlackList extends ConfigurableService
     public const SERVICE_ID = 'tao/ActionBlackList';
     public const OPTION_DISABLED_ACTIONS = 'disabledActions';
     public const OPTION_DISABLED_ACTIONS_FLAG_MAP = 'disabledActionsMap';
+    public const OPTION_ENABLED_ACTIONS_BY_FEATURE_FLAG_MAP = 'enabledActionsByFeatureFlagMap';
```
Comment (bartlomiejmarszal) [inline models/classes/action/ActionBlackList.php:36]: Why do you need this?

**Review by bartlomiejmarszal — CHANGES_REQUESTED** [review]
Why don't you just ad your action to disabled actions

**Review by Karol-Stelmaczonek — APPROVED** [review]
- [x] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful

File: models/classes/action/ActionBlackList.php (hunk 1 above)
Comment (pnal) [reply models/classes/action/ActionBlackList.php:36]: because it should be enabled by feature flag

**Review by tikhanovichA — APPROVED** [review]
- [x] New code is covered by tests (if applicable)
- [x] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful

**Pull request merged**